### PR TITLE
Add protections and dependencies to configuration.sh, system option for grpc.sh, protobuf to grpc.sh modulefile

### PR DESCRIPTION
**Pull request by PascalBoeschoten**
PR for addressing the comments in #897 

1. in grpc.sh you say you depend on protobuf, but then protobuf is not mentioned in the modulefile below. If someone picks up only grpc he will be in trouble.
2. grpc.sh is not pickable from the system, which would be nice.
3. If the above is done, then you need to protect configuration.sh against it.

I think I've done points 1 and 3, but I'm not sure how to do 2.

### CODE REVIEW

**Comment by ktf** [comment]
Please fix the linter.

File: configuration.sh [hunk 1]
```diff
@@ -47,8 +47,8 @@ module load BASE/1.0                                                          \\
             ${BOOST_VERSION:+boost/$BOOST_VERSION-$BOOST_REVISION}            \\
             ${GCC_TOOLCHAIN_ROOT:+GCC-Toolchain/$GCC_TOOLCHAIN_VERSION-$GCC_TOOLCHAIN_REVISION} \\
             ${PROTOBUF_VERSION:+protobuf/$PROTOBUF_VERSION-$PROTOBUF_REVISION} \\
-            grpc/$GRPC_VERSION-$GRPC_REVISION                                 \\
-            Common-O2/$COMMON_O2_VERSION-$COMMON_O2_REVISION
+            ${GRPC_VERSION:+grpc/$GRPC_VERSION-$GRPC_REVISION}                \\
+            ${COMMON_O2_VERSION:+Common-O2/$COMMON_O2_VERSION-$COMMON_O2_REVISION}
```
Comment (ktf) [inline configuration.sh]: Common will always built via AliBuild, no?

Comment (PascalBoeschoten) [reply configuration.sh]: I'm not sure. Should I change it back?

Comment (ktf) [reply configuration.sh]: Yes, I think so.

**Comment by ktf** [comment]
Regarding 2, you can have a look at `prefer_system` and `prefer_system_check` [in the reference manual](http://alisw.github.io/alibuild/reference.html).

**Comment by PascalBoeschoten** [comment]
Is the prefer_system stuff OK like this?

**Comment by ktf** [comment]
Thanks, however maybe you should change the title of the PR to reflect the change? I will than squash all your commits and merge.

**Pull request merged**